### ISSUE TITLE
`array-bracket-newline`: add `consistent` object option

### ISSUE THREAD

**Issue by susnux** [eslint-stylistic/eslint-stylistic#761]
### Clear and concise description of the problem

I want to allow this:
```ts
const a = ['1']
const b = [
    '2',
]
const c = [
    '3',
    '4',
]
```

But not:
```ts
const wrong = ['3',
    '4']
```

### Suggested solution

Similar to `array-element-newline` which allows this we could add a new option to the object config, so it would look like:

```js
'@stylistic/array-bracket-newline': [
	'error',
	{
		consistent: true,
		multiline: true,
	},
]
```

so it would always enforce a new line if it is on a multiline array. But also make it consistent.

### Additional context

basically I want to allow single elements to be on a new line, if I like to do so, but never allow array brackets on the same line as array element in multi-line array.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue
